starter-code/src/QueueDataStructure.js

```diff
@@ -1,2 +1,42 @@
 function QueueDataStructure () {
+  this.queueControl=[];
+  this.MAX_SIZE=8;
+
+  this.isEmpty = function(){
+    if(this.queueControl.length>0){
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  this.canEnqueue = function(){
+    if(this.queueControl.length<this.MAX_SIZE){
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  this.enqueue = function(e) {
+    
+    if (this.canEnqueue){
```

Comment:
Teresa, no te está haciendo esta comprobación porque this.canEnqueue es un método y no una propiedad, entonces nunca entra al if... por tanto tendrías que llamar a la función `if(this.canEnqueue()){...}`